assets/src/scripts/components/hydrograph/legend.js

```diff
@@ -1,188 +1,164 @@
 // functions to facilitate legend creation for a d3 plot
+const memoize = require('fast-memoize');
 const { createSelector } = require('reselect');
+
+const { CIRCLE_RADIUS, MARGIN } = require('./layout');
 const { defineLineMarker, defineCircleMarker, defineRectangleMarker, rectangleMarker } = require('./markers');
-const { CIRCLE_RADIUS } = require('./layout');
-const { MASK_DESC } = require('./timeseries');
+const { lineSegmentsSelector, MASK_DESC, HASH_ID} = require('./timeseries');
 
 
 /**
- * Create a simple horizontal legend
+ * Create a simple legend
  *
- * @param svg
- * @param legendMarkers
- * @param width
- * @param startingXPosition
```

Comment:
I removed all of the positioning parameters. The layout is pretty specific to what we are doing here. I tried playing around with values with not so nice results. Put the ones we still needed as function const.

Reply:
Sounds good.